code/Forms/UsedOnTable.php

```diff
@@ -70,30 +74,119 @@ public function getSchemaDataDefaults()
     public function usage(HTTPRequest $request)
     {
         $record = $this->getRecord();
-        $usage = $record->findOwners();
+        $dataObjects = $record->findOwners();
+        $siteTreeSubClasses = ClassInfo::subclassesFor(SiteTree::class);
 
-        $this->extend('updateUsage', $usage, $record);
+        // Example:
+        // public function updateUsage(ArrayList &$dataObjects, DataObject &$record)
+        //     $dataObjects = $dataObjects->exclude('ClassName', MyDataObject::class);
+        $this->extend('updateUsage', $dataObjects, $record);
 
-        $usageData = [];
+        // TODO: this is all pretty messy, needs a good tidy up before merging
+        $tmpArr = [];
         /**
-         * @var string $userId
-         * @var DataObject $user
+         * @var int $id
+         * @var DataObject $dataObject
          */
-        foreach ($usage as $userId => $user) {
-            $state = $this->getState($user);
-            $link = $user->hasMethod('CMSEditLink') ? $user->CMSEditLink() : null;
-
-            $usageData[] = [
-                'id' => $userId,
-                'title' => $user->getTitle(),
-                'type' => $user->singular_name(),
-                'state' => $state,
+        foreach ($dataObjects as $dataObject) {
+            //$state = $this->getState($user);
+            $link = $dataObject->hasMethod('CMSEditLink') ? $dataObject->CMSEditLink() : null;
+
+            $isPage = false;
+            $parentID = 0;
+            $parentClass = '';
+            if ($dataObject->ParentID && $parent = $dataObject->Parent()) {
+                $parentID = $parent->ID;
+                $parentClass = get_class($parent);
+            }
+            $ancestorPageID = 0;
+            $object = $dataObject;
+            // maximum of 3 levels
+            for ($i = 0; $i < 3; $i++) {
+                $isPage = in_array(get_class($object), $siteTreeSubClasses);
+                if ($isPage) {
+                    if ($object !== $dataObject) {
+                        $ancestorPageID = $object->ID;
+                    }
+                    break;
+                }
+                // extension
+                if ($object->ParentID && $parent = $object->Parent()) {
+                    $object = $parent;
+                    continue;
+                }
+                break;
+            }
+
+            $arr = [
+                'id' => $dataObject->ID,
+                'title' => $dataObject->getTitle() ?: _t(__CLASS__ . '.UNTITLED', 'Untitled'),
+                'type' => ucwords($dataObject->singular_name()),
+                //'state' => $state,
                 'link' => $link,
+                'class' => get_class($dataObject),
+                'parentID' => $parentID,
+                'parentClass' => $parentClass,
+                'ancestorPageID' => $ancestorPageID,
+                'isPage' => $isPage,
             ];
+            $this->extend('updateUsageArray', $arr, $dataObject);
```

Comment:
This whole approach of looking `->Parent()` may be a little pointless in that there's not many (any?) non-page DataObjects that follow this pattern (at least in kitchen-sink).  It may be more sense to not look for `->Parent()` by default and instead rely on this new extension hook.

Reply:
What about `File` and `Folder` or `TaxonomyTerm`? Those come to mind as using the `Hierarchy` extension which provides has_one `Parent`.